lib/collection/src/shards/replica_set/update.rs

```diff
@@ -277,18 +277,29 @@ impl ShardReplicaSet {
         drop(local);
 
         let total_results = all_res.len();
+        let (successes, failures): (Vec<_>, Vec<_>) = all_res.into_iter().partition_result();
 
+        // Determine minimum required successes, never higher than maximum possible successes
+        // Requests having precondition failures on replica states participating in shard transfers
+        // are subtracted because data consistency will be guaranteed by the transfer itself.
+        let pre_condition_fail_count = failures
+            .iter()
+            .filter(|(_, err)| err.is_pre_condition_failed())
+            .filter(|(peer_id, _)| {
+                self.peer_state(peer_id)
+                    .map_or(false, ReplicaState::is_partial_or_recovery)
+            })
+            .count();
         let write_consistency_factor = self
             .collection_config
             .read()
             .await
             .params
             .write_consistency_factor
             .get() as usize;
-
-        let minimal_success_count = write_consistency_factor.min(total_results);
-
-        let (successes, failures): (Vec<_>, Vec<_>) = all_res.into_iter().partition_result();
+        let minimal_success_count = write_consistency_factor
+            .min(total_results - pre_condition_fail_count)
+            .max(1);
```

Comment:
Adjusted write consistency factor handling: do not consider precondition failures on replicas in a shard transfer state, and never allow the factor to go below 1.

Please review this part with care, in case there's some hidden side effect.